students/ZidanLuo/session08/Circle_Class.py

```diff
@@ -0,0 +1,93 @@
+from math import pi
+
+class Circle:
+    def __init__(self, radius):
+        self.radius = radius
+        
+    @property
+    def diameter(self):
+        return self.radius * 2
+
+    @diameter.setter
+    def diameter(self, value):
+        self.radius = value / 2
+
+    @property
+    def area(self):
+        return pi * (self.radius ** 2)
+
+
+    def __str__(self):
+        return "Circle with radius: {0:.6f}".format(self.radius)
+
+
+    def __repr__(self):
+        return f"Circle({self.radius})"
+
+
+    @classmethod
+    def from_diameter(Circle, value):
```

Comment:
we usually use a generic name, like "cls" or "klass" here.

In this case, it is a Circle, but if you subclass, then it will not be anymore -- it could be a Sphere for instance.